assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -0,0 +1,307 @@
+              pickedBuildFromNamespace: buildFrom.namespace || $scope.projectName,
+              pickedBuildFromImageStream: buildFrom.name.split(":")[0],
+              pickedBuildFromImageStreamTag: buildFrom.name.split(":")[1],
+
+              pickedPushToNamespace: buildConfig.spec.output.to.namespace || buildConfig.metadata.namespace,              
+              pickedPushToImageStream: buildConfig.spec.output.to.name.split(":")[0],
+              pickedPushToImageStreamTag: buildConfig.spec.output.to.name.split(":")[1],
+
+              pickedBuildFromDockerImage: $filter('buildStrategy')($scope.updatedBuildConfig).from.name || "",
+            };
+
+            $scope.options.forcePull = false;
+            if ($scope.buildStrategy.forcePull == true) {
+              $scope.options.forcePull = true;
+            }
+
+            if ($scope.buildConfig.spec.strategy.type === "Docker") {
+              $scope.options.noCache = false;
+              if ($scope.buildConfig.spec.strategy.dockerStrategy.noCache === true) {
+                $scope.options.noCache = true;
+              };
+            }
+
+            $scope.buildFrom.projects = ["openshift"];
+            DataService.list("projects", $scope, function(projects) {
+              var projects = projects.by("metadata.name");
+              for (var name in projects) {
+                $scope.buildFrom.projects.push(name);
+                $scope.pushTo.projects.push(name);
+              }
+              $scope.updateImageStreams($scope.options.pickedBuildFromNamespace, true, true);
+              $scope.updateImageStreams($scope.options.pickedPushToNamespace, false, true);
+            });
+
+
+            $scope.loaded = true;
+            // If we found the item successfully, watch for changes on it
+            watches.push(DataService.watchObject("buildconfigs", $routeParams.buildconfig, context, function(buildConfig, action) {
+              if (action === "DELETED") {
+                $scope.alerts["deleted"] = {
+                  type: "warning",
+                  message: "This build configuration has been deleted."
+                };
+              }
+              $scope.buildConfig = buildConfig;
+            }));
+          },
+          // failure
+          function(e) {
+            $scope.loaded = true;
+            $scope.alerts["load"] = {
+              type: "error",
+              message: "The build configuration details could not be loaded.",
+              details: "Reason: " + $filter('getErrorDetails')(e)
+            };
+          }
+        );
+      }));
+
+    // updateImageStreams creates/updates the list of imageStreams and ImageStreamTags from picked namespace. As parameter takes
+    // the picked namespace, buildFrom as a boolean that indicates whether the lists are for the builder image 'picker' or the 
+    // output imageStream 'picker'.
+    $scope.updateImageStreams = function(projectName, buildFrom, firstRun) {
```

Comment:
I was thinking of splitting the `updateImageStreams` into two separate function but that would cause code dups. but I guess its ok in this case since it would make it more clearer. Thanks !